src/pages/dev/general-message-passing/cosmos-gmp.mdx

```diff
@@ -164,4 +163,28 @@ pub fn multi_send_to_evm(
 ## Chain Support
 For a Cosmos blockchain to support receiving GMP messages (naively or via CosmWasm), it should implement the appropriate middleware in Go as part of the consensus layer.
 
-See our [Sample GMP Middleware](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/native-integration/sample-middleware).
\ No newline at end of file
+If you have CosmWasm on your Cosmos chain and want to be able to receieve and deliver messages, install [Osmosis IBC-hooks](https://github.com/osmosis-labs/osmosis/tree/main/x/ibc-hooks).
+
+If you don’t support CosmWasm on your Cosmos chain, you'll need to implement and install your own Axelar GMP module to receive and execute messages. See our [sample reference implementation](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/native-integration/sample-middleware).
+
+
+## Establishing a Path of Trust
+Because IBC messages are permissionless, you'll want to ensure you verify the source of all messages.
+
+You receive a packet on Cosmos chain, you need to verify: 
+1. It came from Axelar (verify IBC channel; Axelar GMP sender account). 
+    How do you specify what Axelar IBC channel is? You whitelist the channel hash. 
+
+    * **[CosmWasm](https://github.com/osmosis-labs/osmosis/blob/main/x/ibc-hooks/README.md#cosmwasm-contract-execution-format)**
+        
+        `Bech32(Hash("ibc-wasm-hook-intermediary" || channelID to Axelar || Axelar GMP account))`  assuming the channel is fixed, which is usually true for mainnet
```

Comment:
Is this ensuring that the channel is correct? A bit unclear what we're using this here for